### PR TITLE
Project api integration

### PR DESCRIPTION
**Pull request by vishaljcj**
---
name: Milestone 4: Collection Management Workflow
about: Changes related to the newly decided workflow for ABM collection Management. Versions would now have projects rather than commits.
---

This can be tested with milestone4_migration branch in  the backend.

**List of functionalities contained in this pull request**
1. Create new collection projects from search page ( instead of commits from crawlers )
2. Add projects to the existing collection
3. Delete project from a collection ( for private versions )
4. Derive version: name and derivedFrom parent is stored

Attached below is the screenshot of all passing test cases.
image

### CODE REVIEW

**Comment by ankur-1989** [comment]
@vishaljcj 
Please check the conflicts.

**Comment by mac24079** [comment]
@vishal. Can you look for this?

I have opened an issue #311  and found out that your PR might help in solving the issue in the develop branch Please let me know in case you need any help.

Thanks.

**Comment by vishaljcj** [comment]
@iamebuka This PR has now been merged with the Milestone 3 changes. It can be tested with the milestone4_migration branch in the backend.

PFB screenshot of passing test cases
image

File: src/app/search/search.component.html [hunk 1]
```diff
@@ -135,24 +135,25 @@ <h5>{{ resultDataSource.data.length }} Result(s)</h5>
   <!-- Selected project list -->
   <div style="max-height:1000px;overflow:auto">
     <h5 class="float-left" style="margin-top:0px" *ngIf="getTotalItems() > 0">
-      {{ getTotalItems() | projectCount }} Selected
+      <!-- {{ getTotalItems() | projectCount }} Selected -->
     </h5>
 
     <div class="col-lg-12 text-right box-padding" *ngIf="getTotalItems() > 0">
-      <button mat-raised-button class="btn btn-success btn-mrgn btn-sm align-middle" (click)="addAll()">
+      <button mat-raised-button class="btn btn-success btn-mrgn btn-sm align-middle" *ngIf="!isFromOtherPage()" (click)="addAll()">
```
Comment (iamebuka) [inline src/app/search/search.component.html]: @vishaljcj  Could you disable the button if no private collections exist as it throws an error. Also, uncertain behaviour. occasionally adds the project to public collections.

Here i created a collection with 3 versions and made them all public
Screenshot from 2019-03-14 15-09-37

Here is a screenshot showing them all public
Screenshot from 2019-03-14 15-11-14

here i tried to add a new project to an existing collection and it shows me public collection
Screenshot from 2019-03-14 15-11-35

.

File: src/app/edit-collection/edit-collection.component.html [hunk 2]
```diff
@@ -31,37 +32,36 @@
...
     <div id="not-frozen" *ngIf="!version.frozen" style="height:28px">
-      <button class="btn btn-primary btn-sm btn-mrgn" mat-raised-button [disabled]="!version.privateStatus" (click)="addProject()">Add
+      <button class="btn btn-primary btn-sm btn-mrgn" mat-raised-button [disabled]="!version.privateStatus" (click)="addProject(version)">Add
         Project</button>
 
       <button mat-raised-button [disabled]="loading || saving || versions.length <= 1 || !version.privateStatus" class="btn btn-sm btn-danger btn-mrgn"
         (click)="removeVersion(version.id)">
         <i class="glyphicon glyphicon-trash"></i> Delete this version</button>
-      <button mat-raised-button [disabled]="loading || saving || !version.privateStatus" class="btn btn-sm btn-danger btn-mrgn"
+      <button mat-raised-button [disabled]="loading || saving || isPublic()" class="btn btn-sm btn-danger btn-mrgn"
```
Comment (iamebuka) [inline src/app/edit-collection/edit-collection.component.html]: I noticed when you derive a version from another version and delete the parent version, the link is lost. Could you look into that also?

File: src/app/search/search.component.html (hunk 1 above)
Comment (vishaljcj) [reply src/app/search/search.component.html]: this issue has been fixed now

File: src/app/edit-collection/edit-collection.component.html (hunk 2 above)
Comment (vishaljcj) [reply src/app/edit-collection/edit-collection.component.html]: right now, code changes have been made in the backend to throw an error if the parentversion doesnt exist in the DB. corresponding frontend changes have been done in this branch. but now you will have to use addexception branch in the backend for testing this.(This addexception branch is in sync with develop, only one LOC has been added, so will raise a PR for merging this. But meanwhile you can use this branch for testing this.

**Pull request merged**